### PR TITLE
Add the `creds_envvar()` function

### LINKED ISSUE #149: feature-request - password as function parameter
Hey is there any way to make it so smtp passwords can be supplied to the function directly/ That way local sending or sending on RS Connect via environment variables would become possible without having to manually enter the password into the prompt.

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `creds_envvar()` credential helper function. It's a variation on `creds()` (manual specification of values). The difference here is that the password can be specified in terms of an environment variable, using `Sys.getenv()` internally to obtain the password value. The required value for the `pass_envvar` arg puts it at the top of the argument signature.

Fixes #149

### CODE REVIEW

File: R/creds_helpers.R [hunk 1]
```diff
@@ -21,6 +21,13 @@
 #' file stored on disk. We can create that file using the
 #' [create_smtp_creds_file()] function.
 #'
+#' The [creds_envvar()] credential helper is a variation on [creds()] where
+#' manual specification of values is needed. The difference here is that the
+#' password can be specified in terms of an environment variable (internally,
+#' [Sys.getenv()] is used with `pass_envvar` to obtain the password). If using
+#' environment variables for all parameters, one can use [Sys.getenv()] directly
+#' for all relevant arguments except for `pass_envvar`.
```
Comment (jcheng5) [inline R/creds_helpers.R]: Suggested rewording:

> The [creds_envvar()] credential helper reads the password from the `SMTP_PASSWORD` environment variable (or an environment variable name that you specify). If using environment variables for other parameters, call `Sys.getenv` manually (e.g. `user = Sys.getenv("SMTP_USER")`).

File: R/creds_helpers.R [hunk 2]
```diff
@@ -31,6 +38,9 @@
 #'   and `port` parameters are the address and port for the SMTP server;
 #'   `use_ssl` is an option as to whether to use SSL: supply a `TRUE` or `FALSE`
 #'   value.
+#' @param pass_envvar An environment variable that holds the value for an email
```
Comment (jcheng5) [inline R/creds_helpers.R]: > The name of the environment variable that...

File: R/creds_helpers.R (hunk 1 above)
Comment (rich-iannone) [reply R/creds_helpers.R]: Thanks. Made this change.

File: R/creds_helpers.R (hunk 2 above)
Comment (rich-iannone) [reply R/creds_helpers.R]: Now changed.

**Comment by mraess** [comment]
Hey Rich,

I tried the `creds_envvar()` function but I'm getting this error even when supplying the password directly into the function and not via `Sys.getenv()` 

> Show in New WindowClear OutputExpand/Collapse Output
> Error: The environment variable defined by `pass_envvar` doesn't exist.

This is my code:

`blastula::creds_envvar(user = Sys.getenv("email_user"), host = "smtp.sendgrid.net", port = 587, 
                       pass_envvar = Sys.getenv("email_pass"))`

Is there anything else I need to be doing?

**Comment by jcheng5** [comment]
Try `pass_envvar="email_pass"`

**Comment by mraess** [comment]
That worked! Thanks a lot!

**Comment by mraess** [comment]
Oh one more question - is there a way to do custom css for the emails, i.e. change background color.

**Pull request merged**